### PR TITLE
Add unit tests for `HeightMapShape3D`

### PR DESCRIPTION
**Pull request by LeonStansfield**
### Description

This PR adds unit tests for the HeightMapShape3D class, as part of '[TRACKER] Unit tests to add or improve #43440'
#43440

### Changes

I have added 7 new tests:

 - Constructor
 -  set_map_width and get_map_width
 - set_map_depth and get_map_depth
 - set_map_data and get_map_data
 - get_min_height
 - get_max_height
 - update_map_data_from_image

### CODE REVIEW

File: tests/test_main.cpp [hunk 1]
```diff
@@ -110,6 +110,7 @@
 #include "tests/scene/test_curve_2d.h"
 #include "tests/scene/test_curve_3d.h"
 #include "tests/scene/test_gradient.h"
+#include "tests/scene/test_height_map_shape_3d.h"
```
Comment (AThousandShips) [inline tests/test_main.cpp:113]: This needs to be moved down under `#ifndef _3D_DISABLED`

Comment (LeonStansfield) [reply tests/test_main.cpp]: Sorted this :smiley:

File: tests/scene/test_height_map_shape_3d.h [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+TEST_CASE("[SceneTree][HeightMapShape3D] get_min_height") {
+	Ref<HeightMapShape3D> height_map_shape = memnew(HeightMapShape3D);
+	height_map_shape->set_map_data(Vector<real_t>{ 1.0, 2.0, 0.5 });
+	CHECK(height_map_shape->get_min_height() == 0.0);
+}
+
+TEST_CASE("[SceneTree][HeightMapShape3D] get_max_height") {
+	Ref<HeightMapShape3D> height_map_shape = memnew(HeightMapShape3D);
+	height_map_shape->set_map_data(Vector<real_t>{ 1.0, 2.0, 0.5 });
+	CHECK(height_map_shape->get_max_height() == 0.0);
```
Comment (Calinou) [inline tests/scene/test_height_map_shape_3d.h:84]: I'm surprised `get_max_height()` is `0.0` here. Shouldn't it be `2.0`?

Comment (LeonStansfield) [reply tests/scene/test_height_map_shape_3d.h]: Ah yes, that is true. After a bit of digging I found the cause was not updating the map width and depth before to accommodate for the map data. I have committed a fix for this now.

File: tests/scene/test_height_map_shape_3d.h [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	map_data.push_back(2.0);
+	height_map_shape->set_map_data(map_data);
+	CHECK(height_map_shape->get_map_data().size() == 4.0);
+	CHECK(height_map_shape->get_map_data()[0] == 0.0);
+	CHECK(height_map_shape->get_map_data()[1] == 0.0);
+}
+
+TEST_CASE("[SceneTree][HeightMapShape3D] get_min_height") {
+	Ref<HeightMapShape3D> height_map_shape = memnew(HeightMapShape3D);
+	height_map_shape->set_map_data(Vector<real_t>{ 1.0, 2.0, 0.5 });
```
Comment (AThousandShips) [inline tests/scene/test_height_map_shape_3d.h:77]: The same change for `get_max_height` should be made here, and the value adjusted

File: tests/scene/test_height_map_shape_3d.h [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	height_map_shape->set_map_depth(1);
+	height_map_shape->set_map_data(Vector<real_t>{ 1.0, 2.0, 0.5 });
+	CHECK(height_map_shape->get_max_height() == 2.0);
+}
+
+TEST_CASE("[SceneTree][HeightMapShape3D] update_map_data_from_image") {
+	// Create a HeightMapShape3D instance.
+	Ref<HeightMapShape3D> height_map_shape = memnew(HeightMapShape3D);
+
+	// Create a mock image with FORMAT_R8 and set its data
```
Comment (AThousandShips) [inline tests/scene/test_height_map_shape_3d.h:95]:
```suggestion
	// Create a mock image with FORMAT_R8 and set its data.
```

File: tests/scene/test_height_map_shape_3d.h [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	Vector<uint8_t> image_data;
+	image_data.push_back(0);
+	image_data.push_back(128);
+	image_data.push_back(255);
+	image_data.push_back(64);
+
+	Ref<Image> image = memnew(Image);
+	image->set_data(2, 2, false, Image::FORMAT_R8, image_data);
+
+	// Call the function with the mock image
```
Comment (AThousandShips) [inline tests/scene/test_height_map_shape_3d.h]:
```suggestion
	// Call the function with the mock image.
```

File: tests/scene/test_height_map_shape_3d.h [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+	// Check the map data.
+	Vector<real_t> expected_map_data = { 0.0, 5.0, 10.0, 2.5 };
+	Vector<real_t> actual_map_data = height_map_shape->get_map_data();
+	real_t tolerance = 0.1;
+
+	for (int i = 0; i < expected_map_data.size(); ++i) {
+		CHECK(Math::abs(actual_map_data[i] - expected_map_data[i]) < tolerance);
+	}
+
+	// Check the min and max heights
```
Comment (AThousandShips) [inline tests/scene/test_height_map_shape_3d.h]:
```suggestion
	// Check the min and max heights.
```

**Review by AThousandShips — APPROVED** [review]
LGTM

File: tests/scene/test_height_map_shape_3d.h [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+	CHECK(height_map_shape->get_map_data()[0] == 0.0);
+	CHECK(height_map_shape->get_map_data()[1] == 0.0);
+}
+
+TEST_CASE("[SceneTree][HeightMapShape3D] get_min_height") {
+	Ref<HeightMapShape3D> height_map_shape = memnew(HeightMapShape3D);
+	height_map_shape->set_map_width(3);
+	height_map_shape->set_map_depth(1);
+	height_map_shape->set_map_data(Vector<real_t>{ 1.0, 2.0, 0.5 });
+	MESSAGE("min height: ", height_map_shape->get_min_height());
```
Comment (kleonc) [inline tests/scene/test_height_map_shape_3d.h:79]: Looks like leftover.
```suggestion
```

**Comment by kleonc** [comment]
Please squash commits into one, see [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) (and ensure the resulting commit has a short descriptive title).

**Comment by AThousandShips** [comment]
Please fix the commit title here as well

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**